libcloud/compute/drivers/kubevirt.py

```diff
@@ -1022,3 +1096,143 @@ def _to_node(self, vm, is_stopped=False):
+        type  node: libcloud `Node` class
+
+        param ports: a list of dictionaries with keys --> values:
+                     'port' --> port to be exposed on the service
+                     'target_port' --> port on the pod/node, optional
+                                       if empty then it gets the same
+                                       value as 'port' value
+                     'protocol' ---> either 'UDP' or 'TCP', defaults to TCP
+                     'name' --> A name for the service
+                     If ports is an empty `list` and a service exists of this
+                     type then the service will be deleted.
+        type  ports: `list` of `dict` where each `dict` has keys --> values:
+                     'port' --> `int`
+                     'target_port' --> `int`
+                     'protocol' --> `str
+                     'name' --> str
+
+        param service_type: Valid types are ClusterIP, NodePort, LoadBalancer
+        type  service_type: `str`
+
+        param cluster_ip: This can be set with an IP string value if you want
+                          manually set the service's internal IP. If the value
+                          is not correct the method will fail, this value can't
+                          be updated.
+        type  cluster_ip: `str`
+
+        param override_existing_ports: Set to True if you want to delete the
+                                       existing ports exposed by the service
+                                       and keep just the ones declared in the
+                                       present ports argument.
+                                       By default it is false and if the
+                                       service already exists the ports will be
+                                       added to the existing ones.
+        type  override_existing_ports: `boolean`
+        '''
+        # check if service exists first
+        namespace = node.extra.get('namespace', 'default')
+        service_name = 'service-{}-{}'.format(service_type.lower(), node.name)
+        service_list = self.ex_list_services(namespace=namespace,
+                                             service_name=service_name)
+
+        ports_to_expose = []
+        # if ports has a falsey value like None or 0
+        if not ports:
+            ports = []
+        for port_group in ports:
+            if not port_group.get('target_port', None):
+                port_group['target_port'] = port_group['port']
+            if not port_group.get('name', ""):
+                port_group['name'] = 'port-{}'.format(port_group['port'])
+            ports_to_expose.append(
+                {'protocol': port_group.get('protocol', 'TCP'),
+                 'port': int(port_group['port']),
+                 'targetPort': int(port_group['target_port']),
+                 'name': port_group['name']})
+        headers = None
+        data = None
+        if len(service_list) > 0:
```

Comment:
Per my comment above, I think it would be better to be explicit and have another ``ex_delete_service`` method.